plasmapy/analysis/magnetic_skeleton.py

```diff
@@ -0,0 +1,1056 @@
+    them.
+
+    Parameters
+    ----------
+    eigen_vals: array_like
+        The eigen values of the Jacobian matrix stored
+        in a 1 by 3 array
+
+    eigen_vectors: array_like
+        The eigen vectors of the Jacobian matrix, each
+        of which are represented by a 1 by 3 array,
+         stored in a 1 by 3 array.
+
+    Returns
+    -------
+    array_like
+        A 1 by 2 numpy array which separately stores the
+        two eigen vectors that define the fan, and the third
+        eigen value which defines the direction of the spine.
+
+    """
+    lambda1_vector = None
+    lambda2_vector = None
+    lambda3_vector = None
+    if (
+        np.isreal(eigen_vals[0])
+        and np.sign(np.real(eigen_vals[0])) * np.sign(np.real(eigen_vals[1])) < 0
+        and np.sign(np.real(eigen_vals[0])) * np.sign(np.real(eigen_vals[2])) < 0
+    ):
+        lambda1_vector = eigen_vectors[:, 1]
+        lambda2_vector = eigen_vectors[:, 2]
+        lambda3_vector = eigen_vectors[:, 0]
+    elif (
+        np.isreal(eigen_vals[1])
+        and np.sign(np.real(eigen_vals[1])) * np.sign(np.real(eigen_vals[0])) < 0
+        and np.sign(np.real(eigen_vals[1])) * np.sign(np.real(eigen_vals[2])) < 0
+    ):
+        lambda1_vector = eigen_vectors[:, 0]
+        lambda2_vector = eigen_vectors[:, 2]
+        lambda3_vector = eigen_vectors[:, 1]
+    elif (
+        np.isreal(eigen_vals[2])
+        and np.sign(np.real(eigen_vals[2])) * np.sign(np.real(eigen_vals[0])) < 0
+        and np.sign(np.real(eigen_vals[2])) * np.sign(np.real(eigen_vals[1])) < 0
+    ):
+        lambda1_vector = eigen_vectors[:, 0]
+        lambda2_vector = eigen_vectors[:, 1]
+        lambda3_vector = eigen_vectors[:, 2]
+
+    if np.iscomplex(lambda1_vector).any():
+        lambda2_vector = np.imag(lambda1_vector)
+        lambda1_vector = np.real(lambda1_vector)
+    return np.array([[lambda1_vector, lambda2_vector], lambda3_vector])
+
+
+# Approximates the field at any point using tl approximation
+def B_approx(vspace, loc):
+    r"""
+    Return the coordinates of a null point within
+    a given grid cell in a vector space using the
+    Newton-Rapshon method.
+
+    Parameters
+    ----------
+    vspace: array_like
+        The vector space as constructed by the vector_space function which is
+        A 1 by 3 array with the first element containing the coordinates,
+        the second element containing the vector values,
+        and the third element containing the delta values for each dimension.
+
+    loc: array_like
+        The location, represented by a 1 by 3 array, in 3D space,
+        for which the magnetic field values will be approximated
+        using trilinear approximation.
+
+    Returns
+    -------
+    array_like
+        A 1 by 3 numpy array which represents the trilinearly
+        approximated vector field value calculated at the given
+        location.
+
+    """
+    # Find the cell
+    # Maybe use numpy.argwhere or numpy.where
```

Comment:
Good catch that you could use those! I'll fire up a debugger by Sunday (a bit tired atm) and try to optimize that for you.